Packages/com.unity.inputsystem/InputSystem/Actions/InputActionParameters.cs

```diff
@@ -0,0 +1,805 @@
+                    var processor = m_State.processors[m_ProcessorCurrentIndex];
+                    if (FindParameter(processor))
+                        return true;
+                }
+                return false;
+            }
+
+            private bool FindParameter(object instance)
+            {
+                if (m_ObjectType != null && !m_ObjectType.IsInstanceOfType(instance))
+                    return false;
+
+                var field = instance.GetType().GetField(m_ParameterName,
+                    BindingFlags.Instance | BindingFlags.Public | BindingFlags.IgnoreCase);
+                if (field == null)
+                    return false;
+
+                m_CurrentParameter = field;
+                m_CurrentObject = instance;
+
+                return true;
+            }
+
+            public bool MoveNext()
+            {
+                while (true)
+                {
+                    if (m_MayBeInteraction && MoveToNextInteraction())
+                        return true;
+
+                    if (m_MayBeProcessor && MoveToNextProcessor())
+                        return true;
+
+                    if (!MoveToNextBinding())
+                        return false;
+
+                    if (m_MayBeComposite && m_CurrentBindingIsComposite)
+                    {
+                        var compositeIndex = m_State.GetBindingState(m_BindingCurrentIndex).compositeOrCompositeBindingIndex;
+                        var composite = m_State.composites[compositeIndex];
+                        if (FindParameter(composite))
+                            return true;
+                    }
+                }
+            }
+
+            public unsafe void Reset()
+            {
+                m_CurrentObject = default;
+                m_CurrentParameter = default;
+                m_InteractionCurrentIndex = default;
+                m_InteractionEndIndex = default;
+                m_ProcessorCurrentIndex = default;
+                m_ProcessorEndIndex = default;
+                m_CurrentBindingIsComposite = default;
+                if (m_MapIndex < 0)
+                {
+                    m_BindingCurrentIndex = -1; // Account for first MoveNext().
+                    m_BindingEndIndex = m_State.totalBindingCount;
+                }
+                else
+                {
+                    m_BindingCurrentIndex = m_State.mapIndices[m_MapIndex].bindingStartIndex - 1; // Account for first MoveNext().
+                    m_BindingEndIndex = m_State.mapIndices[m_MapIndex].bindingStartIndex + m_State.mapIndices[m_MapIndex].bindingCount;
+                }
+            }
+
+            public Parameter Current => new Parameter
+            {
+                instance = m_CurrentObject,
+                field = m_CurrentParameter,
+                bindingIndex = m_BindingCurrentIndex,
+            };
+
+            object IEnumerator.Current => Current;
+
+            public void Dispose()
```

Comment:
Intentionally empty dispose?

Reply:
Yup. Mandated by interface but there's nothing to dispose here.

Reply:
ok, yeah I didn't say anything to get rid of either unless nulling references was needed but I guess not